### ISSUE TITLE
Padding Token?

### ISSUE THREAD

**Issue by IsNoobgrammer** [Locutusque/TPU-Alignment#2]
So much Thanks for putting it all together 

so for Mistral Architechure pad_token is present as `'<unk>'` and eos_token is `"</s>"` so by setting 
![image](https://github.com/Locutusque/TPU-Alignment/assets/112808251/c21dadce-5b0f-4940-94b9-e0549c116924)
will tokenizer.pad_token=tokenizer.eos_token for mistral/other architechture  not effect model performance ??

and doubt on my side shouldn't  FLAGS['STEPS'] = len(dataset)/batch_size 

Can i Add PR for LoRA ? (currently working on that)

Can you look on [this](https://www.kaggle.com/code/defdet/llama-7b-fine-tuning-on-tpu) its implementing model without sharding for every node/device a model 
 and [derived-this](https://www.kaggle.com/code/sandiago21/llm-science-exam-llms-training-tpu)

**Comment by Locutusque** [comment]
Thanks for pointing this out. I'll try removing the tokenizer.pad_token = tokenizer.eos_token and see if model performance increases, but it really shouldn't do anything. Also the number of steps is correctly calculated later. I'll also try using the distributed sampler and see if that improves memory efficiency.

**Comment by Locutusque** [comment]
Fixed the oversight I made with the Distributed Sampler, it has sped up training time a lot.

**Comment by IsNoobgrammer** [comment]
> Fixed the oversight I made with the Distributed Sampler, it has sped up training time a lot. I'll commit the changes tonight.

Can't we natively implement it in PyTorch Lightning  ? (Since we already have Training loop and Optimizer configured in Pytorch)
 Then it will be easy to scale for any architechture and can be easily used with other Fine-Tuning Techiniques (LoRA , Quantization etc.. ) and Lighting also allows us to scale to TPU/GPU per requirements 
 
 I am working on it .. will keep updating 
 P.S : I am currently new to this ..  so might take some/(more than expected) time
